### ISSUE TITLE
adding an ImageDataset class

### ISSUE THREAD

**Issue by eminorhan** [AI-HHMI/miao#17]
2D models are often strong baselines in 3D domains. Would it be useful to write a new `ImageDataset` class along the lines of `VolumeDataset` to make it easier to train both 2D and 3D models with `miao`? It could support the same random and sequential sampling strategies as `VolumeDataset`, as well as the same cheap data augmentations (axis aligned rotations and flips).

**Comment by eminorhan** [comment]
I think for this use case, creating three separate "degenerate" `VolumeDataset`s with shapes `(1,p,p)`, `(p,1,p)` and `(p,p,1)` and then concatenating them using the `ConcatDataset` machinery from `torch.utils.data` may be a better option on second thoughts, something like this:
```python
cfg_z = MiaoConfig(..., patch_size=[1, P, P], ...)
cfg_y = MiaoConfig(..., patch_size=[P, 1, P], ...)
cfg_x = MiaoConfig(..., patch_size=[P, P, 1], ...)
ds_z, ds_y, ds_x = VolumeDataset(cfg_z), VolumeDataset(cfg_y), VolumeDataset(cfg_x)
concat_ds = ConcatDataset([ds_z, ds_y, ds_x])
```
I just checked that this seems to work, but it requires a custom `collate_fn` that removes the singleton dimensions when creating a corresponding `DataLoader`.

I think the refactor involved in adding a bespoke `ImageDataset` would complicate the code base too much and make it more difficult to maintain.

**Comment by jakobtroidl** [comment]
Ok, sounds good! I think it could make sense to add an example section in the README or an example notebook, that illustrates how one can use miao for 2D data ? Could you add such an example ?
